packages/docusaurus-theme-classic/src/theme/Layout/index.js

```diff
@@ -36,6 +37,10 @@ function Layout(props) {
   const metaImage = image || defaultImage;
   const metaImageUrl = siteUrl + useBaseUrl(metaImage);
   const faviconUrl = useBaseUrl(favicon);
+  const hasBlog = presets.some(preset => (preset[1] || {}).blog !== undefined);
+  const rssFeedLink = siteUrl + useBaseUrl('blog/feed.xml');
+  const atomFeedLink = siteUrl + useBaseUrl('blog/atom.xml');
```

Comment:
I do not like to initialize these variables even when there is no blog exists, but hooks cannot be called conditionally. What are some ideas to fix this?